pkg/v1/remote/options.go

```diff
@@ -16,14 +16,34 @@ package remote
 
 import (
 	"context"
+	"crypto/tls"
+	"net"
 	"net/http"
+	"time"
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/logs"
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
 )
 
+// DefaultTransport is a an extension of the default net/http DefaultTransport with http2 disabled.
+// According to moby/buildkit#1420, net/http lack of tunable flow control prevents full throughput on push.
+var DefaultTransport = &http.Transport{
```

Comment:
I'm hesitant to just expose this as a global. In buildkit it looks like they create a new one of these whenever it's needed, so I'm worried that this isn't safe to reuse.

Reply:
Yeah, I found that odd since the `net/http` one is a statically defined like here. Happy to switch it up.

Reply:
Let's add a DefaultTransport function to pkg/internal that returns a new one of these (like they do in buildkit) and use that instead of this global. I don't want to commit to exposing this as a public API, and I assume they're constructing a new one each time for a good reason.

Reply:
I guess that won't work for cmd/crane/cmd/root.go but I think it's fine to just copy a private implementation into that package for now.

Reply:
Cool, added to both cmd/crane and `pkg/internal/net` (honestly not sure what to call it, I didn't want to shadow `pkg/v1/remote/transport`).

Reply:
> honestly not sure what to call it

`net`'s fine -- internal names don't matter to much.